components/HolidayList/index.js

```diff
@@ -0,0 +1,36 @@
+import { useEffect, Fragment } from 'react'
+import PropTypes from 'prop-types'
+import { Spinner } from 'react-bootstrap';
+
+const HolidayList = ({ country, year, holidays, fetchHolidays, loadingHolidays }) => {
+
+  useEffect(() => {
+    if(country && year) {
+      fetchHolidays(country, year)
+    }
+  }, [country, year]);
+
+  if(loadingHolidays) {
```

Comment:
This is right. But I want global loader which will act in following ways.

1. When user opens app user will see the loader which will cover full page.
2. Based on what page you are trying to access your code should make api request or get data from local storage.
e.g. _On landing page your app will fetch countries only and once country data is received loader will go away. But on holidays page first your app will make call to countries then to holidays api and once data for both is fetched loader will go away._

Reply:
I understand using different component for loader but why cover full page with loader ? That just makes it annoying to user as the entire page flickers when there is an API request. Why not just put loader on the part that is updating

Reply:
I think flickering is something which can be handle in react. It won't be smooth if not implemented correctly. We faced the same issue on Google it. Current loader is better than the one we created one but not very smooth. check the toggle full page loader for example.

Also, moving loader globally will add extra complexity to the app where you need to handle the code based on the responses of multiple api which including handling failed api requests

Reply:
moved to main app.